src/basic/fs-util.c

```diff
@@ -742,17 +742,26 @@ bool unsafe_transition(const struct stat *a, const struct stat *b) {
 }
 
 static int log_unsafe_transition(int a, int b, const char *path, unsigned flags) {
-        _cleanup_free_ char *n1 = NULL, *n2 = NULL;
+        _cleanup_free_ char *n1 = NULL, *n2 = NULL, *user_a = NULL, *user_b = NULL;
+        struct stat stat_a, stat_b;
 
         if (!FLAGS_SET(flags, CHASE_WARN))
                 return -ENOLINK;
 
         (void) fd_get_path(a, &n1);
         (void) fd_get_path(b, &n2);
 
+        if (fstat(a, &stat_a) < 0)
+                return -errno;
```

Comment:
We shouldn't miss useful messages if some parts of it can't be put together. Hence, please do the opposite here, and do uid_to_name() only if the fstat succeeds, but proceed ignoring it otherwise.